app/Http/Controllers/CommunityController.php

```diff
@@ -40,8 +45,42 @@ public function getChat()
         return view('community.chat');
     }
 
-    public function getLive()
+    public function getLive(Request $request)
     {
-        return view('community.live');
+        $streams = null;
+        $featuredStream = null;
+        $streams = Cache::remember('livestreams', 5, function () {
+            $justin_api_url = 'https://api.twitch.tv/kraken/streams?on_site=1&limit=40&offset=0&game=Osu!';
+            $data = json_decode(file_get_contents($justin_api_url));
+            $streams = $data->streams;
+
+            return $streams;
+        });
+
+        if (Cache::has('featuredStream')) {
```

Comment:
why not just do the `::get` and check if it's null? Probably saves one memcache trip.